python/pyspark/sql/datasource.py

```diff
@@ -332,8 +331,13 @@ def partitions(self) -> Sequence[InputPartition]:
             message_parameters={"feature": "partitions"},
         )
 
+    if TYPE_CHECKING:
+        from pyarrow import RecordBatch
+    else:
+        RecordBatch = None
```

Comment:
How about let's move this to the beginning of this file? Then we don't have to import this multiple times.
```
if TYPE_CHECKING:
    from pyarrow import RecordBatch
    from pyspark.sql.session import SparkSession
```